### PR TITLE
Places API & Config

### PR DESCRIPTION
**Pull request by Bottigerd**
### Changes:
- API calls no longer occur on launch, but the map is center on the current location if there are location permissions
- Broke up the location authorization/permission function into multiple ones so hopefully the code is cleaner like Jeff wanted
- Places API is implemented, if reverse-geocoding returns a `place_id` in the struct, it'll call the places API and return the name of the place with the town, state, postal code, and country to the address still mimics the formatted address response of the reverse geocoding api
- Implemented `config.json` to store sensitivity information like the API key. It's in the `.gitignore` already. The config goes in `location-app/SwiftUI-UserLocation/config.json`. You can find the config in the google drive for safe-keeping

### Something I noticed and will need fixing and was present before this branch:
- the API calls run but don't necessarily save the response before the application tries to display the address. This means the button is pressed about three times before the right information is shown
- I believe this has to do with async and sync functions or completion handlers (ways to make sure other code doesn't execute until something else is done)
- I spent way too long debugging this using completion handlers and async functions but could not figure it out. Async functions don't look promising but completion handlers look a bit more promising
- This is a plea for help for someone to also take a look at this

### CODE REVIEW

File: SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift [hunk 1]
```diff
@@ -258,20 +263,24 @@ final class ContentViewModel: NSObject, ObservableObject,
...
             } else if (address_component.types.contains("postal_code")) {
-                name = name + " " + address_component.shortName
+                postal_code = address_component.shortName
             } else if (address_component.types.contains("country")) {
-                name = name + ", " + address_component.shortName
+                country = address_component.shortName
             }
         }
-        return name
+        return place_name + ", " + locality + ", " + admin_area_1 + " " + postal_code + ", " + country
```
Comment (kittyTyree) [inline SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift:283]: I am getting an error on this line: "The compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions"

File: SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift [hunk 2]
```diff
@@ -258,6 +257,23 @@ final class ContentViewModel: NSObject, ObservableObject,
         return coordinates2d.latitude.description + "," + coordinates2d.longitude.description
     }
     
+    internal func getPlaceName() -> String {
```
Comment (kittyTyree) [inline SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift:265]: This version of getPlaceName() does not cause a compiler error.

Comment (kittyTyree) [reply SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift:265]: ^ and by this version I mean the version that got changed

**Review by kittyTyree — CHANGES_REQUESTED** [review]
The changes for a commit that got pushed that cuts down lines of code in getPlaceName() causes a compiler error when I try and run the code. Even though the formatting gets changed and, granted, looks cleaner in the code, I might suggest changing it back to what it was before the changes were made so that it does not run this compiler issue (even if the error didn't show up on your system.)

File: SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift (hunk 2 above)
Comment (Bottigerd) [reply SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift:265]: I made getPlaceName() so that the given address when a place has a place ID is more helpful, otherwise its something like "Google Headquarters 6" or "118 Ellis St" and nothing else. I'll try to fix the compiler error so we can keep that helpful-ness. 

Do you know what coordinates you were using on your simulator when you got that error so I can recreate it?

Comment (Bottigerd) [reply SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift:265]: I split up the line that adds the different pieces of the place name and gave default values to the variables. I tried testing different locations to see if I could prompt the error again but I wasn't able to. So, either it's fixed or I still haven't found that edge case. Could you test again?

Comment (alsheikha) [reply SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift:265]: Your change worked for me! Gonna look at some of the other issues now

**Review by kittyTyree — APPROVED** [review]

File: SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift (hunk 2 above)
Comment (kittyTyree) [reply SwiftUI-UserLocation/SwiftUI-UserLocation/ContentViewModel.swift:265]: Worked no issues!

**Review by alsheikha — APPROVED** [review]
returnString solution worked for me. Address still doesn't update on each click
